dune-project

```diff
@@ -41,3 +41,17 @@
    (and
     (>= 2.4.1)
     :with-doc))))
+
+(package
+ (name multicore-magic-dscheck)
+ (synopsis
+  "A implementation of multicore-magic API using the atomic module of DScheck to make DScheck tests possible in libraries using multicore-magic")
+ (allow_empty)
+ (depends
+  (ocaml
+   (>= 4.12.0))
+  ;; Test dependencies
+  (multicore-magic
+   (>= 2.2.0))
+  (dscheck
+   (= dev))))
```

Comment:
BTW, when using `pin-depends` it is not necessary to specify `dev` as the version as the `pin-depends` kind of overwrites that.  So, you could already change this to refer to the future version of `dscheck` that is about to be published.